src/core/build/geometryUtils.js

```diff
@@ -57,12 +57,18 @@ export function ensureIndex( geo, options ) {
 //                      g1 = [16, 40]           g2 = [41, 60]
 //
 // we would need four BVH roots: [0, 15], [16, 20], [21, 40], [41, 60].
-export function getFullGeometryRange( geo ) {
+export function getFullGeometryRange( geo, range ) {
 
 	const triCount = getTriCount( geo );
-	const drawRange = geo.drawRange;
-	const start = drawRange.start / 3;
-	const end = ( drawRange.start + drawRange.count ) / 3;
+	let workingRange = geo.drawRange;
+	if ( range ) {
+
+		workingRange = getRangesIntersection( workingRange, range );
```

Comment:
I think it's okay if we just use the passed in "range" here instead of trying to find the intersection of drawRange and the passed in range. It's simpler and in this way a user can override the use of draw range if needed, as well.

That means we'd also have to default the "range" option to null. Or we could default "range" to the geometries draw range to be more clear. What do you think?